go/vt/vttablet/tabletmanager/tm_state.go

```diff
@@ -243,13 +245,19 @@ func (ts *tmState) UpdateTablet(update func(tablet *topodatapb.Tablet)) {
 	ts.publishForDisplay()
 }
 
-func (ts *tmState) updateLocked(ctx context.Context) {
+func (ts *tmState) updateLocked(ctx context.Context) error {
 	span, ctx := trace.NewSpan(ctx, "tmState.update")
 	defer span.Finish()
 	ts.publishForDisplay()
-
+	// SetServingType can result in error. Although we have forever retries to fix these transient errors
+	// but, under certain condition these errors are non-transient (see https://github.com/vitessio/vitess/issues/10145).
+	// There is no way to distinguish between retry (transient) and non-retryable errors, therefore we will
+	// always return error from 'SetServingType' and 'applyDenyList' to our client. It is up to them to handle it accordingly.
+	// UpdateLock is called from 'ChangeTableType', 'Open' and 'RefreshFromTopoInfo'. For 'Open' and 'RefreshTopoInfo' we don't need
+	// to propagate error to client hence no changes there but we will propagate error from 'ChangeTableType' to client.
+	var returnErr error
 	if !ts.isOpen {
-		return
+		return fmt.Errorf("TabletManager state is not open")
```

Comment:
This change should be reverted. In the code path where `RefreshFromTopoInfo` is called, `isOpen` is false.
This is most likely what is causing the CI test failures.